.github/workflows/master_dev_checks.yml

```diff
@@ -0,0 +1,409 @@
+            echo "//${{env.NEXUS_URL}}/repository/${{env.NEXUS_REPO_NPM}}:always-auth = true" >> .npmrc
+            echo "@tessell:registry=https://nexus.tessell.cloud/repository/${{env.NEXUS_REPO_NPM}}" >> .npmrc
+            cat .npmrc
+
+        - name: Install Packages
+          run: |
+            npm cache clean --force
+            if [ ! -d "node_modules/@octokit/rest" ]; then
+              npm install @octokit/rest --legacy-peer-deps
+            fi
+            sudo sudo npm install jsonlint -g
+            npm install node-fetch js-yaml
+            
+        - name: Check for Jsonlint
+          run: |
+            set -e
+            git diff --name-only --diff-filter=AM ${{ github.event.pull_request.base.sha }}...${{ github.event.pull_request.head.sha }} | grep '\.json$' | xargs -I{} sh -c 'jsonlint -q "{}" || { echo "::error::Invalid JSON file: {}"; exit 1; }'
+
+        - name: Check for merge conflicts
+          uses: actions/github-script@v6.4.1
+          with:
+            github-token: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}
+            script: |
+              head_commit = "${{ github.sha }}"
+              base_commit = "${{ github.sha }}"
+              if ( "${{ github.event.before }}" != '0000000000000000000000000000000000000000') { 
+                base_commit = "${{ github.event.before }}"
+              }
+              if ( "${{ github.event_name }}" == 'pull_request') {
+                console.log("Using the base branch's commit for comparing.")
+                base_commit = "${{ github.event.pull_request.base.sha  }}"
+              }
+              response = await github.rest.repos.compareCommits({
+                  owner: "${{ github.repository_owner }}",
+                  repo: "${{  github.repository }}".split('/')[1],
+                  head: head_commit,
+                  base: base_commit
+              });
+              response.data.files.forEach(function(file_entry) {
+                console.log(file_entry.filename);
+                dir_name = file_entry.filename.split("/")[0];
+                console.log(file_entry.patch);
+                if(typeof file_entry.patch !== 'undefined'){
+                  if ( file_entry.patch.includes("+<<<<<<<") || file_entry.patch.includes("+=======") || file_entry.patch.includes("+>>>>>>>")) {
+                    core.setFailed("Please resolve Merge Conflict in: " + dir_name );  
+                  }
+                }
+                else{
+                  console.log("Skipping:" + file_entry.filename);
+                }
+              });
+              console.log("No merge conflicts found"); 
+        
+        - name: Check DB Migration Scripts
+          id: check_db_migration
+          uses: actions/github-script@v6.4.1
+          if: github.event_name == 'pull_request'
+          with:
+            github-token: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}
+            script: |
+              head_commit = "${{ github.sha }}"
```

Comment:
Line 274 - 290 are same as 237 to 251;  Can we make it as one step with `response` saved as output of that step and use it as needed.